cincinnati/src/lib.rs

```diff
@@ -216,12 +216,29 @@ impl Graph {
                     if release.manifestref().unwrap_or(&missing_manifest_ref)
                         != node.manifestref().unwrap_or(&missing_manifest_ref)
                     {
-                        bail!(
-                            "mismatched manifest ref for concrete release {}: {}, {}",
-                            release.version(),
-                            release.manifestref().unwrap_or(&missing_manifest_ref),
-                            node.manifestref().unwrap_or(&missing_manifest_ref)
-                        )
+                        let no_arch = "none".to_string();
+                        let release_arch = release
+                            .get_metadata_mut()
+                            .unwrap()
+                            .get("release.openshift.io/architecture")
+                            .unwrap_or(&no_arch);
+                        let node_arch = node
+                            .get_metadata_mut()
+                            .unwrap()
+                            .get("release.openshift.io/architecture")
+                            .unwrap_or(&no_arch);
+
+                        if release_arch == node_arch {
```

Comment:
I'm not clear on how this whole `Conrete` block avoids comparing the single-arch amd64 and s390x, etc. releases for the same version today.  Is there some by-arch lookup higher in the stack?

Reply:
the `find_by_version()` which this code snippet is a part of uses version+arch to get to this block. 
so when we have two same versions+arch here, if its an abstract release it gets replaced by concrete release. 
But if there is concrete release already in the graph, we were checking for integrity before and bailing out on sha sum mismatch. 
now the new code block checks the `release.openshift.io/architecture` field and if both are different, it ignores the ones with `multi` set in `release.openshift.io/architecture`. 
So effectively we are ignoring single-arch images built for multi-arch clusters and keeping single arch releases specifically built for single-arch. 
the manifest list images wont be affected here because sha sum will be same and if it differs, it should bailout like it does today